server/src/main/java/org/elasticsearch/common/util/CollectionUtils.java

```diff
@@ -47,6 +44,27 @@ public static boolean isEmpty(Object[] array) {
         return array == null || array.length == 0;
     }
 
+    public static <T> void unique(List<T> list, Comparator<T> cmp) {
+        if (list.size() <= 1) {
+            return;
+        }
+
+        int prevNdx = 0;
+        T prevValue = list.get(0);
+        for (int i = 1; i < list.size(); ++i) {
+            T nextValue = list.get(i);
+            if (cmp.compare(nextValue, prevValue) != 0 && prevNdx++ != i) {
```

Comment:
Do we need to pre-increment rather than post-increment? Otherwise it looks to me like a list where all elements are unique would still overwrite all the time?

Reply:
You are right, it should be pre-increment.